### PR TITLE
Adding particle package in utils

### PR DESCRIPTION
**Pull request by stefanopolitano**
This PR contains:

- addition of "particle" package (https://github.com/scikit-hep/particle)
- modification of particle mass getter function

### CODE REVIEW

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @stefanopolitano! I was wondering whether we need these functions at all, wouldn't it be sufficient to call `Particle.from_pdgid(pdg_id).mass` or `Particle.from_name(pdg_name).mass` in the `set_particle_mass` method here https://github.com/flarefly/flarefly/blob/7c571e1bc11ff5b2ebbc6aa8f3a64368472233b5/flarefly/fitter.py#L781? You can move the `mass` parameter in the `kwargs` and add the `pdg_code` and the `name` and then use the corresponding setter if one or the other is set

**Comment by stefanopolitano** [comment]
Hi @fgrosa, I implemented this function as a substitute of the previous one but indeed, we can just use the particle package in the fitter. 
Regarding the requirements I forgot: I will add it. Thanks!

**Comment by fgrosa** [comment]
> Hi @fgrosa, I implemented this function as a substitute of the previous one but indeed, we can just use the particle package in the fitter. Regarding the requirements I forgot: I will add it. Thanks!

Yes I would simply put it in the fitter, since the functions provided by the `particle` package already give us all that we need 👍

**Comment by stefanopolitano** [comment]
Hi @fgrosa, I think I'm getting the same error you got with macos

File: flarefly/fitter.py [hunk 1]
```diff
@@ -778,24 +779,36 @@ def get_significance(self, idx=0, nsigma=3):
 
         return significance, significance_err
 
-    def set_particle_mass(self, idx, mass, **kwargs):
+    def set_particle_mass(self, idx, pdg_id=-999, pdg_name='', **kwargs):
```
Comment (fgrosa) [inline flarefly/fitter.py:782]:
```suggestion
    def set_particle_mass(self, idx, **kwargs):
```

File: flarefly/fitter.py [hunk 2]
```diff
@@ -778,24 +779,36 @@ def get_significance(self, idx=0, nsigma=3):
...
         Parameters
         -------------------------------------------------
         idx: int
             Index of the signal
-        mass: float
-            The mass to be set
+        pdg_id: int
+            PDG ID of the particle
+        pdg_name: str
+            Name of the particle
```
Comment (fgrosa) [inline flarefly/fitter.py:793]:
```suggestion
```

File: flarefly/fitter.py [hunk 3]
```diff
@@ -778,24 +779,36 @@ def get_significance(self, idx=0, nsigma=3):
...
             - fix: bool
                 fix the mass parameter
         """
         mass_name = 'm' if self._name_signal_pdf_[idx] == 'cauchy' else 'mu'
+        if 'mass' in kwargs:
+            mass = kwargs['mass']
+        elif pdg_id != -999:
+            mass = Particle.from_pdgid(pdg_id).mass*1e-3
+        elif pdg_name:
+            mass = Particle.from_name(pdg_name).mass*1e-3
```
Comment (fgrosa) [inline flarefly/fitter.py:811]:
```suggestion
        if 'mass' in kwargs:
            mass = kwargs['mass']
        elif 'pdg_id' in kwargs:
            mass = Particle.from_pdgid(kwargs['pdg_id']).mass*1e-3
        elif 'pdg_name' in kwargs:
            mass = Particle.from_name(kwargs['pdg_name']).mass*1e-3
        else:
            Logger(f'"mass", "pdg_id", and "pdg_name" not provided, mass value for signal {idx} will not be set',
                         'ERROR')
```

File: flarefly/fitter.py [hunk 4]
```diff
@@ -778,24 +779,36 @@ def get_significance(self, idx=0, nsigma=3):
...
+        pdg_id: int
+            PDG ID of the particle
+        pdg_name: str
+            Name of the particle
         **kwargs: dict
             Additional optional arguments:
+            - mass: float
+                The mass of the particle
+            - mass_name: str
+                name of the mass variable
```
Comment (fgrosa) [inline flarefly/fitter.py:799]:
```suggestion
            - mass: float
                The mass of the particle
            - pdg_id: int
                PDG ID of the particle (alternative to mass)
            - pdg_name: str
                Name of the particle (alternative to mass)
```

File: tests/test_massfitter_binned.py [hunk 5]
```diff
@@ -17,8 +17,8 @@
 FITTERBINNED = F2MassFitter(DATABINNED,
                             name_signal_pdf=['gaussian', 'gaussian'],
                             name_background_pdf=['expo'])
-FITTERBINNED.set_particle_mass(0, 1.872, fix=True)
-FITTERBINNED.set_particle_mass(1, 2.010, limits=[2.000, 2.020])
+FITTERBINNED.set_particle_mass(0, mass=1.872, fix=True)
+FITTERBINNED.set_particle_mass(1, mass=2.010, limits=[2.000, 2.020])
```
Comment (fgrosa) [inline tests/test_massfitter_binned.py:21]: Let's test also a different argument
```suggestion
FITTERBINNED.set_particle_mass(1, pdg_id=413, limits=[2.000, 2.020])
```

**Review by fgrosa — CHANGES_REQUESTED** [review]
Thanks @stefanopolitano! I just suggested you a small modification to have all the possible arguments in the `kwargs`

**Comment by fgrosa** [comment]
> Hi @fgrosa, I think I'm getting the same error you got with macos

Now it breaks on wheel installation, it might be that we have to modify a bit the `setup.py`, but let's do it in another PR (I opened an issue https://github.com/flarefly/flarefly/issues/30)

**Review by fgrosa — APPROVED** [review]

File: tests/test_massfitter_binned.py [hunk 6]
```diff
@@ -17,8 +17,8 @@
 FITTERBINNED = F2MassFitter(DATABINNED,
                             name_signal_pdf=['gaussian', 'gaussian'],
                             name_background_pdf=['expo'])
-FITTERBINNED.set_particle_mass(0, 1.872, fix=True)
-FITTERBINNED.set_particle_mass(1, 2.010, limits=[2.000, 2.020])
+FITTERBINNED.set_particle_mass(0, mass=1.872, fix=True)
+FITTERBINNED.set_particle_mass(1, pdg_id=413, limits=[2.000, 2.020])
 FITTERBINNED.set_signal_initpar(0, 'sigma', 0.010)
 FITTERBINNED.set_signal_initpar(1, 'sigma', 0.015)
```
Comment (fgrosa) [inline tests/test_massfitter_binned.py:23]: To make the fit converge and pass the test
```suggestion
FITTERBINNED.set_signal_initpar(1, 'sigma', 0.015, limits=[0.01, 0.03])
```

**Review by fgrosa — CHANGES_REQUESTED** [review]

**Review by fgrosa — APPROVED** [review]

**Pull request merged**